### PR TITLE
Various nixos/gitlab fixes

### PR DESCRIPTION
**Pull request by nightkr**
###### Motivation for this change

Currently Gitlab doesn't even launch properly, because a few paths aren't set up properly. This patch fixes that.

It also disables automatic installation of `pg_trgm` if the server isn't set to `127.0.0.1` (localhost), since the `psql` calls aren't aware of remote servers (see #41476).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by teozkr** [comment]
With these patches it launches and I am able to log in, but actions that require gitaly (such as creating a repository) don't work, because it's unable to connect to the gitaly UDS.

**Comment by gbuisson** [comment]
If that can help, I noticed gitaly isn't working to import repos because it seems that rake is not bundled with it.

**Comment by krav** [comment]
The issue with Gitaly is that ruby-cd is not wrapped, adding 
`  buildInputs = [rubyEnv.wrappedRuby];`
fixes the problem

File: nixos/modules/services/misc/gitlab.nix [hunk 1]
```diff
@@ -560,6 +560,7 @@ in {
         mkdir -p ${cfg.statePath}/tmp/sockets
         mkdir -p ${cfg.statePath}/shell
         mkdir -p ${cfg.statePath}/db
+        mkdir -p ${cfg.statePath}/uploads
```
Comment (globin) [inline nixos/modules/services/misc/gitlab.nix]: Are you sure that this is necessary, as far as I had understood `uploads.storage_path = cfg.statePath;` in the config above should have made it possible yo remove this. How did you trigger an error?

**Comment by Denommus** [comment]
I was trying to deploy gitlab today and also got this problem during migration: https://gist.github.com/Denommus/fb595242a9a3f3796b0198d0da56572d#file-journalctl

**Comment by Denommus** [comment]
It seems that with these fixes, somehow, my error also doesn't happen.

**Pull request merged**